benchmarks/functional_autograd_benchmark/utils.py

```diff
@@ -0,0 +1,59 @@
+from collections import defaultdict
+
+# Utilities to make nn.Module functional
+def del_attr(obj, names):
+    if len(names) == 1:
+        delattr(obj, names[0])
+    else:
+        del_attr(getattr(obj, names[0]), names[1:])
+def set_attr(obj, names, val):
```

Comment:
It took a few minutes to understand what was going on in `del_attr` and `set_attr`. Maybe they should be renamed to `del_nested_attr` and `set_nested_attr`. Some docstrings explaining what they are doing would be nice too ("del_attr deletes the attribute specified by object.{".".join(names)}")

Proposal: All callsites of `delattr` and `setattr` look like the following:
```
del_attr(mod, name.split("."))
set_attr(mod, name.split("."), p)
```
It might be easier for `del_attr` to take a string representation of the attribute to be deleted (e.g. "foo.bar") rather than a list.

Reply:
Yes, I made it this way to keep these two functions as simple as possible as they are recursive. But I can have an extra check if the name is a string.